policy/compiler.go

```diff
@@ -84,17 +100,28 @@ func (v *CompiledVariable) Declaration() *decls.VariableDecl {
 // CompiledMatch represents a match block which has an optional condition (true, by default) as well
 // as an output or a nested rule (one or the other, but not both).
 type CompiledMatch struct {
+	exprID     int64
 	cond       *cel.Ast
 	output     *OutputValue
 	nestedRule *CompiledRule
 }
 
+func (m *CompiledMatch) SourceID() int64 {
+	return m.exprID
+}
+
 // Condition returns the compiled predicate expression which must evaluate to true before the output
 // or subrule is entered.
 func (m *CompiledMatch) Condition() *cel.Ast {
 	return m.cond
 }
 
+// ConditionIsLiteral indicates whether the condition for the match is a literal with a given value.
+func (m *CompiledMatch) ConditionIsLiteral(val ref.Val) bool {
+	c := m.cond.NativeRep().Expr()
```

Comment:
This may panic due to `cond` being `nil` if we fail to compile the condition for a match: https://github.com/google/cel-go/pull/991/files#diff-329a773e11a74db809329cb56d571b04b394de161cef34e6ac66f9bf2fd21c45R256

Reply:
On a compiled match, the condition is always non-nil since it's either trivially `true` or a custom expression. On a parsed match, the value can be `nil`, but there are better guarantees here.

Reply:
If the custom expression cannot be compiled, we'll enter this literal check when doing the reachable check, through `HasOptionalOutput()` function, at which point the condition should be `nil`.

I think this will panic on an input like:

```
rule:
  match:  
    - condition: bad_condition
      output: "'foo'"
```

Reply:
Ah, good point. I realized the `cel.Ast` wasn't as `nil`-safe as it could be, so the check should now work just fine even if `cond` is `nil` given the update to the nil-safety for `cel.Ast`